README.md

```diff
@@ -1,27 +1,508 @@
+    return (
+        <article>
+            <h1>{this.props.title}</h1>
+            <p>{this.props.body}</p>
+        </article>
+    )
+}
+```
+
+which can also be written as an arrow function:
+
+```js
+const BlogPost = props =>
+    <article>
+        <h1>{props.title}</h1>
+        <p>{props.body}</p>
+    </article>
+```
+
+or using destructured props:
+
+```js
+const BlogPost = ({title, body}) =>
+    <article>
+        <h1>{title}</h1>
+        <p>{body}</p>
+    </article>
+```
+
+Because the above component can be written as a single function that doesn't
+rely on any **internal state** or component-lifecycle (more on that later), we
+call those components _pure_ and _stateless_. **Purity** is an important concept
+in programming and not something specific to React. A function can be considered
+to be _pure_ if — and only if — it:
+
+1. always returns the same result when called with the same arguments:
+
+    In other words, `<BlogPost title={'My title'} body={'My body'} />` is always
+    going to _return_ the exact same result. It doesn't matter when or "where"
+    we render the component: The only aspect that the component itself cares
+    about are its props.
+
+2. doesn't have any side effects:
+
+    Typically, your component's `render` function shouldn't have any fancy logic
+    that starts your coffee machine or loads data from your API. In functional
+    programming, when we talk about "side effects", we mean mutation of some
+    form of external state. Sounds fancy, but essentially it means pure
+    functions can't load data from an API or update some global variable (such
+    as `localStorage`).
+
+### Internal State
+
+As we just saw, props can be seen as arguments for pure components. Pure
+components don't rely on any internal state or side-effects: They simply return
+a virtual representation of how they want to "look" like.
+
+There is only one problem: props are always being "passed down" from the
+component's immediate ancestor. So for example, the `Blog` component passes the
+blog post data to the `BlogPost` component. But where does the blog *data*
+actually come from?
+
+```js
+class Blog extends React.Component {
+    render () {
+        return (
+            <div>
+                <BlogPost title={'Hello World!'} body={'Hello, this is my blog.'} />
+                <BlogPost title={'Good bye!'} body={'I\'m busy. I\'m shutting this blog down.'} />
+            </div>
+        )
+    }
+}
+```
+
+In the above example, we simply hard-coded the blog titles and bodies. Obviously
+this isn't really what we want, otherwise we might as well just render
+everything on the server — not very *reactive* at all.
+
+Instead we typically want to dynamically load the data from some API endpoint,
+for instance `GET /api/posts`. So whenever we render a new blog component, we
+do a XHR call and load the blog posts that we want to render.
+
+In React, the function that gets executed once a component has been "rendered"
```

Comment:
You can assume that students will have learned about `componentDidMount()` and `setState()` at this point, so no need to explain those in detail. You can, however, leave in a small blurb to remind them (e.g. 'Remember, `componentDidMount()` will fire once when the component is mounted).